pkg/apis/ceph.rook.io/v1/annotations_test.go

```diff
@@ -80,6 +81,10 @@ func TestCephAnnotationsMerge(t *testing.T) {
 	assert.Equal(t, "crashval", c["crash"])
 	assert.Equal(t, "allval1", c["allkey1"])
 	assert.Equal(t, "allval2", c["allkey2"])
+	d := GetOSDAnnotations(testAnnotations)
+	assert.Equal(t, "allval1", d["allkey1"])
+	assert.Equal(t, "allval2", c["allkey2"])
+	assert.Equal(t, "osdval", d["osdkey"])
```

Comment:
One of the things that I'm seeing is that we don't seem to have a test that ensures a value set in `all` can be overridden by values set in other places. 

While we are fixing the code here, could we also test at least once here to ensure overrides are happening as desired?

For example, what if we add `allval1: osdkey1` to the OSD test? I think we should find that `allval1` == `osdkey1` and not `allkey1`.

Reply:
I agree with you, but according to the `TestAnnotationsMerge` test, `testAnnotationsPart1` represents `all`, while `testAnnotationsPart2` represents specific components (e.g., osd/mgr). `all` takes higher priority.

So my question is: which has higher priority, `all` or specific components (e.g., `osd`/`mgr`)? Should the components override `all`?

Reply:
Normally I would think specific overrides generic, but clearly I am mistaken here. I didn't realize there was a test below part 2.